### PR TITLE
Meta: Add tests for `show-whitespace`

### PR DESCRIPTION
**Pull request by fregante**
- Preparing for #5709

### CODE REVIEW

File: source/helpers/show-whitespace-on-line.test.ts [hunk 1]
```diff
@@ -0,0 +1,224 @@
...
+function assertHighlighted(t: ExecutionContext, actual: string, expected: string): void {
+	t.is(
+		show(actual, false),
+		expected,
+	);
+}
+
+test('showWhiteSpacesOnLine raw', t => {
+	assert(
+		t,
```
Comment (fregante) [inline source/helpers/show-whitespace-on-line.test.ts:58]: I can't find a way to use `t.is` without passing it here. I can't find a way to create custom global assertions either, like:

```js
t.transformsTo(actual, expected)
```

https://jestjs.io/docs/expect#expectextendmatchers

File: source/helpers/show-whitespace-on-line.test.ts [hunk 2]
```diff
@@ -0,0 +1,224 @@
...
+		'',
+		'',
+	);
+	assert(
+		t,
+		' ',
+		'•',
```
Comment (fregante) [inline source/helpers/show-whitespace-on-line.test.ts:65]: They're specifically set vertically to help visually evaluate the substitutions too

File: source/features/show-whitespace.tsx [hunk 3]
```diff
@@ -1,67 +1,20 @@
...
 
 import features from '.';
-import getTextNodes from '../helpers/get-text-nodes';
 import onNewComments from '../github-events/on-new-comments';
 import {onDiffFileLoad} from '../github-events/on-fragment-load';
 import {codeElementsSelector} from '../github-helpers/dom-formatters';
-
-// `splitText` is used before and after each whitespace group so a new whitespace-only text node is created. This new node is then wrapped in a <span>
-function showWhiteSpacesOn(line: Element): void {
-	const shouldAvoidSurroundingSpaces = Boolean(line.closest('.blob-wrapper-embedded')); // #2285
```
Comment (fregante) [inline source/features/show-whitespace.tsx:15]: `showWhiteSpacesOn` was moved without change, except for extracting `shouldAvoidSurroundingSpaces`
